File: 7568.c
#include<stdio.h>

int main(){
    int numOfCases, caseWeight, caseHeight;
    int weight[50] = {0, };
    int height[50] = {0, };
    int result[50] = {0, };
    
    scanf("%d", &numOfCases);
    for(int i = 0; i < numOfCases; i++)
        scanf("%d %d", &weight[i], &height[i]);
        
    for(int i = 0; i < numOfCases; i++){
        int counter = 1;
        for(int j = 0; j < numOfCases; j++){
            if(weight[j] > weight[i] && height[j] > height[i])
                counter++;
        }
        result[i] = counter;
    }
    
    for(int i = 0; i < numOfCases; i ++){
        printf("%d ", result[i]);
    }
    return 0;
}

File: 7568.py
numOfCases = int(input())

weight = list()
height = list()
result = list()

for i in range(numOfCases) :
    caseWeight, caseHeight = map(int, input().split())
    weight.append(caseWeight)
    height.append(caseHeight)

for i in range(numOfCases) :
    counter = 1
    for j in range(numOfCases) :
        if weight[j] > weight[i] and height[j] > height[i] :
            counter = counter + 1
    result.append(counter)

for each in result :
    print(each, end=" ")
